service_crategen/codegen/src/generator/query.rs

```diff
@@ -153,13 +153,15 @@ fn generate_serializer_signature(name: &str, shape: &Shape) -> String {
 }
 
 fn generate_list_serializer(service: &Service, shape: &Shape) -> String {
-
     let member_shape = service.shape_for_member(shape.member.as_ref().unwrap()).unwrap();
     let primitive = member_shape.is_primitive();
-
+    let is_shape_flattened = match shape.flattened {
+        None => false,
+        Some(_) => true,
```

Comment:
Good catch, we should change to that.

`Some(shape_defined_flatness) => shape_defined_flatness,` ?